applications/nrf_desktop/src/modules/dfu.c

```diff
@@ -116,9 +207,14 @@ static void handle_dfu_data(const struct config_event *event)
 	return;
 
 dfu_finish:
-	flash_area_close(flash_area);
-	flash_area = NULL;
 	k_delayed_work_cancel(&dfu_timeout);
+	if (img_length == cur_offset) {
+		flash_area_close(flash_area);
+		flash_area = NULL;
+	} else {
+		erase_offset = (FLASH_PAGE_ID(cur_offset) + 1) << FLASH_PAGE_SIZE_LOG2;
```

Comment:
we just failed to erase or write to page... what's the point of starting background erase if flash is non functioning?

Reply:
Changed as discussed offline. Now erase is only in background and DFU is started only if flash is clean.